diff --git a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/definitions.d.ts b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/definitions.d.ts
index 76f412a..d4d7638 100644
--- a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/definitions.d.ts
+++ b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/definitions.d.ts
@@ -12,6 +12,7 @@ export interface LoginOptions extends BaseOptions {
     /** MSAL always sends the scopes 'openid profile offline_access'.  Do not include any of these scopes in the scopes parameter. */
     scopes?: string[];
     prompt?: 'login' | 'none' | 'consent' | 'create' | 'select_account';
+    isExpired?: boolean;
 }
 export type LogoutOptions = BaseOptions;
 export interface MsAuthPlugin {
diff --git a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.d.ts b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.d.ts
index 8c28773..2a6e539 100644
--- a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.d.ts
+++ b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.d.ts
@@ -5,6 +5,7 @@ interface WebBaseOptions extends BaseOptions {
 }
 interface WebLoginOptions extends WebBaseOptions {
     scopes: string[];
+    isExpired?: boolean;
 }
 type WebLogoutOptions = WebBaseOptions;
 interface AuthResult {
diff --git a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.js b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.js
index ca81dc5..2d22330 100644
--- a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.js
+++ b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/esm/web.js
@@ -4,7 +4,7 @@ export class MsAuth extends WebPlugin {
     async login(options) {
         const context = this.createContext(options);
         try {
-            return await this.acquireTokenSilently(context, options.scopes).catch(() => this.acquireTokenInteractively(context, options.scopes));
+            return await this.acquireTokenSilently(context, options.scopes, options.isExpired).catch(() => this.acquireTokenInteractively(context, options.scopes));
         }
         catch (error) {
             console.error('MSAL: Error occurred while logging in', error);
@@ -49,10 +49,11 @@ export class MsAuth extends WebPlugin {
         });
         return { accessToken, idToken, scopes };
     }
-    async acquireTokenSilently(context, scopes) {
+    async acquireTokenSilently(context, scopes, isExpired) {
         const { accessToken, idToken } = await context.acquireTokenSilent({
             scopes,
             account: context.getAllAccounts()[0],
+            forceRefresh: isExpired || false,
         });
         return { accessToken, idToken, scopes };
     }
diff --git a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.cjs.js b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.cjs.js
index 07e0c5d..c692bde 100644
--- a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.cjs.js
+++ b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.cjs.js
@@ -7,7 +7,7 @@ class MsAuth extends core.WebPlugin {
     async login(options) {
         const context = this.createContext(options);
         try {
-            return await this.acquireTokenSilently(context, options.scopes).catch(() => this.acquireTokenInteractively(context, options.scopes));
+            return await this.acquireTokenSilently(context, options.scopes, options.isExpired).catch(() => this.acquireTokenInteractively(context, options.scopes));
         }
         catch (error) {
             console.error('MSAL: Error occurred while logging in', error);
@@ -52,10 +52,11 @@ class MsAuth extends core.WebPlugin {
         });
         return { accessToken, idToken, scopes };
     }
-    async acquireTokenSilently(context, scopes) {
+    async acquireTokenSilently(context, scopes, isExpired) {
         const { accessToken, idToken } = await context.acquireTokenSilent({
             scopes,
             account: context.getAllAccounts()[0],
+            forceRefresh: isExpired || false,
         });
         return { accessToken, idToken, scopes };
     }
diff --git a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.js b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.js
index ffe409a..f05e973 100644
--- a/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.js
+++ b/node_modules/@recognizebv/capacitor-plugin-msauth/dist/plugin.js
@@ -2,16 +2,16 @@ var capacitorMsAuth = (function (exports, core, msalBrowser) {
     'use strict';
 
     class MsAuth extends core.WebPlugin {
-        async login(options) {
-            const context = this.createContext(options);
-            try {
-                return await this.acquireTokenSilently(context, options.scopes).catch(() => this.acquireTokenInteractively(context, options.scopes));
-            }
-            catch (error) {
-                console.error('MSAL: Error occurred while logging in', error);
-                throw error;
-            }
+    async login(options) {
+        const context = this.createContext(options);
+        try {
+            return await this.acquireTokenSilently(context, options.scopes, options.isExpired).catch(() => this.acquireTokenInteractively(context, options.scopes));
         }
+        catch (error) {
+            console.error('MSAL: Error occurred while logging in', error);
+            throw error;
+        }
+    }
         logout(options) {
             const context = this.createContext(options);
             if (!context.getAllAccounts()[0]) {
@@ -50,13 +50,14 @@ var capacitorMsAuth = (function (exports, core, msalBrowser) {
             });
             return { accessToken, idToken, scopes };
         }
-        async acquireTokenSilently(context, scopes) {
-            const { accessToken, idToken } = await context.acquireTokenSilent({
-                scopes,
-                account: context.getAllAccounts()[0],
-            });
-            return { accessToken, idToken, scopes };
-        }
+    async acquireTokenSilently(context, scopes, isExpired) {
+        const { accessToken, idToken } = await context.acquireTokenSilent({
+            scopes,
+            account: context.getAllAccounts()[0],
+            forceRefresh: isExpired || false,
+        });
+        return { accessToken, idToken, scopes };
+    }
     }
 
     const MsAuthPlugin = core.registerPlugin('MsAuthPlugin', {
diff --git a/node_modules/@recognizebv/capacitor-plugin-msauth/ios/Plugin/Plugin.swift b/node_modules/@recognizebv/capacitor-plugin-msauth/ios/Plugin/Plugin.swift
index b7347bd..d855c27 100644
--- a/node_modules/@recognizebv/capacitor-plugin-msauth/ios/Plugin/Plugin.swift
+++ b/node_modules/@recognizebv/capacitor-plugin-msauth/ios/Plugin/Plugin.swift
@@ -15,6 +15,7 @@ public class MsAuthPlugin: CAPPlugin {
         }
 
         let scopes = call.getArray("scopes", String.self) ?? []
+        let isExpired = call.getBool("isExpired") ?? false
 
         var promptType: MSALPromptType = .selectAccount
         if let prompt = call.getString("prompt")?.lowercased() {
@@ -53,7 +54,7 @@ public class MsAuthPlugin: CAPPlugin {
                 return
             }
 
-            self.acquireTokenSilently(applicationContext: context, scopes: scopes, account: currentAccount, completion: completion)
+            self.acquireTokenSilently(applicationContext: context, scopes: scopes, account: currentAccount, isExpired: isExpired, completion: completion)
         }
     }
 
@@ -281,8 +282,9 @@ public class MsAuthPlugin: CAPPlugin {
         }
     }
 
-    func acquireTokenSilently(applicationContext: MSALPublicClientApplication, scopes: [String], account: MSALAccount, completion: @escaping (MSALResult?) -> Void) {
-        let parameters = MSALSilentTokenParameters(scopes: scopes, account: account)
+    func acquireTokenSilently(applicationContext: MSALPublicClientApplication, scopes: [String], account: MSALAccount, isExpired: Bool, completion: @escaping (MSALResult?) -> Void) {
+      let parameters = MSALSilentTokenParameters(scopes: scopes, account: account)
+      parameters.forceRefresh = isExpired
 
         applicationContext.acquireTokenSilent(with: parameters) { (result, error) in
 
